extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/scan/ScanQueryKit.java

```diff
@@ -185,7 +178,7 @@ public QueryDefinition makeQueryDefinition(
                          .inputs(new StageInputSpec(firstStageNumber))
                          .signature(signatureToUse)
                          .maxWorkerCount(1)
-                         .shuffleSpec(null) // no shuffling should be required after a limit processor.
+                         .shuffleSpec(finalShuffleSpec) // Apply the final shuffling after limit spec.
```

Comment:
Could you please explain the reasoning about this change with the below query shapes. What are the new new stages added . 

* q like ` insert + limit  with partitioned by all`
* q like `insert + limit with partitioned by day` assuming validations on planning side is removed. 
* q like `select * from foo group by a limit 100`-> taskReport
* q like `select * from foo limit 100` -> durableStorage
* q like `select * from foo limit 100 order by x`
* q like `export select * from foo limit 100`
* a inner select query.

Reply:
Updated the PR description